client/new-stack-editor/lib/views/flexsplit/storage.coffee

```diff
@@ -0,0 +1,65 @@
+kd = require 'kd'
+FlexSplit = require './index'
+
+
+module.exports = class FlexSplitStorage extends kd.Object
+
+
+  constructor: (options = {}, data) ->
+
+    super options, data
+
+    @storage    = {}
+    @viewCount  = 0
+    @eventCount = 0
+
+    if adapterClass = @getOption 'adapter'
+      @adapter = new adapterClass
+
+
+  addView: (view, identifier, options = {}) ->
+
+    options.restore          ?= yes
+    options.keepExpandStatus ?= no # Experimental
+
+    view.on FlexSplit.EVENT_RESIZED, (fractions) =>
+      @set identifier, fractions
+      @store()
+
+    if options.keepExpandStatus
+      view.on [
+        FlexSplit.EVENT_EXPANDED
+        FlexSplit.EVENT_COLLAPSED
+      ], (fractions) =>
+
+        @eventCount++
+        @set identifier, fractions
+
+        if @eventCount is @viewCount - 1
+          @eventCount = 0
+          @store()
```

Comment:
Can you explain this part? I don't quite understand what happens if `keepExpandStatus` is `true`.